### PR TITLE
Add api to customize library

### PR DESCRIPTION
**Pull request by haoliu-amp**
- Added unit tests.
- Used demo app testing it's working properly.

### CODE REVIEW

File: src/main/java/com/amplitude/api/AmplitudeClient.java [hunk 1]
```diff
@@ -1017,8 +1041,8 @@ protected long logEvent(String eventType, JSONObject eventProperties, JSONObject
             }
 
             JSONObject library = new JSONObject();
-            library.put("name", Constants.LIBRARY);
-            library.put("version", Constants.VERSION);
+            library.put("name", this.libraryName == null ? Constants.LIBRARY_UNKNOWN : this.libraryName);
```
Comment (djih) [inline src/main/java/com/amplitude/api/AmplitudeClient.java]: nonblocking suggestion: in the `setLibraryName`, if the input value is `null`, then you can do `this.libraryName = LIBRARY_UNKNOWN`, and then you don't have to do this comparison every time

File: src/main/java/com/amplitude/api/AmplitudeClient.java [hunk 2]
```diff
@@ -1017,8 +1041,8 @@ protected long logEvent(String eventType, JSONObject eventProperties, JSONObject
...
+            library.put("version", this.libraryVersion == null ? Constants.VERSION_UNKNOWN : this.libraryVersion);
```
Comment (djih) [inline src/main/java/com/amplitude/api/AmplitudeClient.java]: same here

File: src/main/java/com/amplitude/unity/plugins/AmplitudePlugin.java [hunk 3]
```diff
@@ -94,6 +94,22 @@ public static void enableForegroundTracking(String instanceName, Application app
         Amplitude.getInstance(instanceName).enableForegroundTracking(app);
     }
 
+    public static void enableCoppaControl(String instanceName) {
```
Comment (djih) [inline src/main/java/com/amplitude/unity/plugins/AmplitudePlugin.java]: 👍

File: src/main/java/com/amplitude/api/AmplitudeClient.java (hunk 1 above)
Comment (haoliu-amp) [reply src/main/java/com/amplitude/api/AmplitudeClient.java]: My thought is to leave it as is. Keep it original. 

Both are fine.

**Pull request merged**